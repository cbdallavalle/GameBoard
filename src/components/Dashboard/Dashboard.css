.Dashboard {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px;
}

nav button {
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  background-color: #ffffffcf;
  border-top-left-radius: 13px;
  border-top-right-radius: 13px;
  border: none;
  padding: 10px;
}

nav button:hover,
nav button:focus {
  cursor: pointer;
  background-color: #BD9B89;
}

.Dashboard h2 {
  font-family: 'Righteous', cursive;
  font-size: 25px;
}

nav {
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 630px;
  height: fit-content;
}

.search {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.search button {
  width: 170px;
  height: 42px;
  margin: 0;
  border: 2px solid black;
  font-size: 17px;
  font-family: 'Open Sans', sans-serif;
}

.search input {
  width: 170px;
  height: 35px;
}