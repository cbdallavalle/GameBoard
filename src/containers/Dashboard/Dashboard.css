.Dashboard {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Dashboard h2 {
  font-family: 'Righteous', cursive;
  font-size: 25px;
}

.body-cont {
  background-color: #f4f4f4a6;
  overflow-y: scroll;
  height: 90vh;
}

.search {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.search button {
  width: 170px;
  height: 42px;
  margin: 0;
  border: 2px solid black;
  font-size: 17px;
  font-family: 'Open Sans', sans-serif;
}

.search input {
  width: 170px;
  height: 35px;
}