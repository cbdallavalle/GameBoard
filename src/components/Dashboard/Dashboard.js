import React, { Component } from 'react';
import { CardContainer } from '../CardContainer/CardContainer';
import { searchGames } from '../../actions';
import { connect } from 'react-redux';
import { Nav } from '../Nav/Nav';
import Search from '../Search/Search';
import AddFriends from '../AddFriends/AddFriends';
import { SignOutBtn } from '../SignOutBtn/SignOutBtn';
import './Dashboard.css';

export class Dashboard extends Component {
  constructor() {
    super()
    this.state = {
      display: 'your-games',
    }
  }

  determineDisplay = () => {
    if(this.state.display === 'your-games') {
      return (
        <CardContainer favorites={this.props.favorites}/>
      )  
    } else if (this.state.display === 'friends-games') {
      return (
        <CardContainer favorites={{}}/>
      )
    } else if (this.state.display === 'search-games') {
      return (
        <Search />
      )
    } else if (this.state.display === 'search-friends') {
      return (
        <AddFriends />
      )
    }

  }

  handleClick = e => {
    this.setState({ display: e.target.name})
  }

  render() {
    return (
      <section className="Dashboard">
        <nav>
          <button name='your-games' onClick={this.handleClick}>
            <i className="fas fa-user"></i> Your Games
          </button>
          <button name='friends-games' onClick={this.handleClick}>
            <i className="fas fa-users"></i> Your friend's games
          </button>
          <button name='search-games' onClick={this.handleClick}>
            Search for games 
          </button>
          <button name="search-friends" onClick={this.handleClick}>
            Search for friends
          </button>
        </nav>
        {this.determineDisplay()}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  favorites: state.favorites
})

export default connect(mapStateToProps, null)(Dashboard);