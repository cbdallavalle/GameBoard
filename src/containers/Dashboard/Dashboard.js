import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { db } from '../../firebase';

import AddFriends from '../AddFriends/AddFriends';
import Search from '../Search/Search';
import { CardContainer } from '../../components/CardContainer/CardContainer';
import { Header } from '../../components/Header/Header';
import { Nav } from '../../components/Nav/Nav';

import './Dashboard.css';

export class Dashboard extends Component {
  constructor() {
    super();
    this.state = {
      friendsFavorites: {},
      error: ''
    };
  }

  componentWillReceiveProps = () => {
    this.updateFavorites();
  }

  updateFavorites = async() => {
    try {
      if (this.props.user.uid) {
        const friendsFavorites = await 
          db.getFriendsFavorites(this.props.user.uid);
        this.setState({ friendsFavorites });
      }
    } catch (error) {
      this.setState({ error: error.message });
    }
  }

  render() {
    return (
      <section className="Dashboard">
        <Header />
        <div className="body-cont">
        <Nav handleClick={this.handleClick}/>
          <Switch>
            <Route
              exact path="/dashboard/your-games"
              render={ () => 
                <CardContainer 
                  favorites={this.props.favorites} 
                  type={"games"} 
                /> 
              }
            />
            <Route 
              exact path="/dashboard/friends-games" 
              render={ () => 
                <CardContainer 
                  favorites={this.state.friendsFavorites} 
                  type={"friends"} 
                /> 
              }
            />
            <Route 
              exact path="/dashboard/search-games" 
              component={ Search }
            />
            <Route 
              exact path="/dashboard/search-users" 
              component={ AddFriends }
            />
          </Switch>
        </div>
      </section>
    );
  }
}

Dashboard.propTypes = {
  favorites: PropTypes.object.isRequired,
  user: PropTypes.object
};

export const mapStateToProps = state => ({
  favorites: state.favorites,
  user: state.user
});

export default withRouter(connect(mapStateToProps, null)(Dashboard));