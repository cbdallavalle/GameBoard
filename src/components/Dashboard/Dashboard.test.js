import React from 'react';
import { shallow } from "enzyme";
import { Dashboard } from './Dashboard';

describe('Dashboard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Dashboard />)
    expect(wrapper).toMatchSnapshot();
  })

  // it('should have a default state of empty strings for gameSearch, friendSearch', () => {
  //   const wrapper = shallow(<Dashboard />)
  //   expect(wrapper.state()).toEqual({ gameSearch: '', friendSearch: ''})
  // })

  // it('should change state when handleChange has been called', () => {
  //   const wrapper = shallow(<Dashboard />)
  //   const mockEvent = { target: {name: "gameSearch", target: "Captain Sonar"}}
  //   wrapper.instance().handleChange(mockEvent)
  // })
})

// describe('mapDispatchToProps', () => {
//   it('should map addSearch to props', () => {
//       const mockDispatch = jest.fn();
//       const mapped = mapDispatchToProps(mockDispatch);

//       mapped.addSearch("Evolution");
//       expect(mockDispatch).toHaveBeenCalled();
//   })
// })
