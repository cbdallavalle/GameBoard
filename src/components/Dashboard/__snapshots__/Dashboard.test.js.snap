// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section>
          <h2>
                    <i
                              className="fas fa-user"
                    />
                    Your games
          </h2>
          <Unknown />
          <div
                    className="search"
          >
                    <Link
                              replace={false}
                              to="/search"
                    >
                              <button
                                        id="search-btn"
                              >
                                        Search for games
                              </button>
                    </Link>
          </div>
</section>,
        <Unknown />,
        <section>
          <h2>
                    <i
                              className="fas fa-users"
                    />
                    Your friend's games
          </h2>
          <Unknown />
          <div
                    className="search"
          >
                    <button
                              id="search-btn"
                    >
                              Search for friends
                    </button>
          </div>
</section>,
      ],
      "className": "Dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              <i
                            className="fas fa-user"
              />
              Your games
</h2>,
            <Unknown />,
            <div
              className="search"
>
              <Link
                            replace={false}
                            to="/search"
              >
                            <button
                                          id="search-btn"
                            >
                                          Search for games
                            </button>
              </Link>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-user"
/>,
                "Your games",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-user",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Your games",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/search"
>
                <button
                                id="search-btn"
                >
                                Search for games
                </button>
</Link>,
              "className": "search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  id="search-btn"
>
                  Search for games
</button>,
                "replace": false,
                "to": "/search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Search for games",
                  "id": "search-btn",
                },
                "ref": null,
                "rendered": "Search for games",
                "type": "button",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              <i
                            className="fas fa-users"
              />
              Your friend's games
</h2>,
            <Unknown />,
            <div
              className="search"
>
              <button
                            id="search-btn"
              >
                            Search for friends
              </button>
</div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-users"
/>,
                "Your friend's games",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-users",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Your friend's games",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                id="search-btn"
>
                Search for friends
</button>,
              "className": "search",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search for friends",
                "id": "search-btn",
              },
              "ref": null,
              "rendered": "Search for friends",
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "section",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section>
            <h2>
                        <i
                                    className="fas fa-user"
                        />
                        Your games
            </h2>
            <Unknown />
            <div
                        className="search"
            >
                        <Link
                                    replace={false}
                                    to="/search"
                        >
                                    <button
                                                id="search-btn"
                                    >
                                                Search for games
                                    </button>
                        </Link>
            </div>
</section>,
          <Unknown />,
          <section>
            <h2>
                        <i
                                    className="fas fa-users"
                        />
                        Your friend's games
            </h2>
            <Unknown />
            <div
                        className="search"
            >
                        <button
                                    id="search-btn"
                        >
                                    Search for friends
                        </button>
            </div>
</section>,
        ],
        "className": "Dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                <i
                                className="fas fa-user"
                />
                Your games
</h2>,
              <Unknown />,
              <div
                className="search"
>
                <Link
                                replace={false}
                                to="/search"
                >
                                <button
                                                id="search-btn"
                                >
                                                Search for games
                                </button>
                </Link>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-user"
/>,
                  "Your games",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Your games",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/search"
>
                  <button
                                    id="search-btn"
                  >
                                    Search for games
                  </button>
</Link>,
                "className": "search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    id="search-btn"
>
                    Search for games
</button>,
                  "replace": false,
                  "to": "/search",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Search for games",
                    "id": "search-btn",
                  },
                  "ref": null,
                  "rendered": "Search for games",
                  "type": "button",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                <i
                                className="fas fa-users"
                />
                Your friend's games
</h2>,
              <Unknown />,
              <div
                className="search"
>
                <button
                                id="search-btn"
                >
                                Search for friends
                </button>
</div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="fas fa-users"
/>,
                  "Your friend's games",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-users",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Your friend's games",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  id="search-btn"
>
                  Search for friends
</button>,
                "className": "search",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Search for friends",
                  "id": "search-btn",
                },
                "ref": null,
                "rendered": "Search for friends",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
